In [266]:
import pandas as pd

In [267]:
test_2017 = pd.read_csv('../know_Data/test/KNOW_2017_test.csv')

In [268]:
test_2017

idx  aq1_1 aq1_2  aq2_1 aq2_2  aq3_1 aq3_2  aq4_1 aq4_2  aq5_1  ...  \
0        0      3     4      2     2      3     3      1            3  ...   
1        1      5     5      3     5      5     5      5     5      4  ...   
2        2      5     5      5     4      5     4      1            1  ...   
3        3      4     5      5     6      4     6      3     4      4  ...   
4        4      5     6      4     5      4     5      1            1  ...   
...    ...    ...   ...    ...   ...    ...   ...    ...   ...    ...  ...   
9481  9481      3     4      4     5      4     5      5     6      4  ...   
9482  9482      4     5      4     6      5     6      5     6      5  ...   
9483  9483      3     2      1            2     1      3     3      1  ...   
9484  9484      4     5      3     4      3     4      1            1  ...   
9485  9485      3     4      4     5      4     4      4     5      3  ...   

     bq36  bq37 bq38  bq38_1 bq39_1  bq39_2 bq40  bq41_1 bq41_2  bq41_3  
0       2    26    3     비서학      1       1    1    3000           2300  
1       1    57    4     농화학      1       1    1    5500           2500  
2       1    31    4    신문방송      1       1    1    4300           4000  
3       1    35    6      화학      1       1    1    4100           3000  
4       1    36    4    광고홍보      1       1    1    2800           2000  
...   ...   ...  ...     ...    ...     ...  ...     ...    ...     ...  
9481    1    52    1              2       6                3400          
9482    1    48    5     요업과      1       1    1    7000           2400  
9483    1    44    2     인문계      2       6                4500          
9484    1    44    4   컴퓨터공학      1       1    1    6000           4000  
9485    1    42    3      기계      1       1    1    3000           2000  

[9486 rows x 155 columns]

In [269]:
train_2017 = pd.read_csv('know_2017_enutf.csv')
train_2017

idx  aq1_1  aq1_2  aq2_1  aq2_2  aq3_1  aq3_2  aq4_1  aq4_2  aq5_1  \
0        0      3      3      3      3      3      3      4      4      3   
1        1      4      5      4      5      3      4      3      4      3   
2        2      3      4      3      4      3      4      5      6      4   
3        3      3      3      3      3      3      5      4      5      4   
4        4      4      5      3      4      3      4      4      5      3   
...    ...    ...    ...    ...    ...    ...    ...    ...    ...    ...   
9481  9481      3      5      2      4      3      3      2      2      2   
9482  9482      5      5      5      5      5      5      3      4      4   
9483  9483      3      3      4      6      3      3      4      5      4   
9484  9484      3      5      3      5      4      5      3      4      3   
9485  9485      3      4      3      4      3      4      3      4      3   

      ...  bq37  bq38  bq38_1  bq39_1  bq39_2  bq40  bq41_1  bq41_2  bq41_3  \
0     ...    52     2      실업       1       1     1    4000       0    2200   
1     ...    38     4    건축공학       1       1     1    3900       0    2400   
2     ...    50     4    건축공학       1       1     1    4000       0    2400   
3     ...    42     4    환경학과       1       1     1    7000       0    3500   
4     ...    51     4    건축공학       1       1     1    4000       0    2500   
...   ...   ...   ...     ...     ...     ...   ...     ...     ...     ...   
9481  ...    50     4   산업디자인       1       1     1    5200       0    1800   
9482  ...    37     4  우주항공공학       1       1     1    4000       0    3000   
9483  ...    32     2      기계       1       4     2    2700       0    1500   
9484  ...    40     4   문헌정보학       1       1     1    6800       0    2500   
9485  ...    48     3      전기       1       1     1    4000       0    3700   

      knowcode  
0       825101  
1       140204  
2       140204  
3       140601  
4       140204  
...        ...  
9481    411301  
9482    151105  
9483    701101  
9484     25402  
9485     15201  

[9486 rows x 156 columns]

## 이상치 파악

In [270]:
test_2017.info(verbose=1)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9486 entries, 0 to 9485
Data columns (total 155 columns):
 #    Column  Dtype  
---   ------  -----  
 0    idx     int64  
 1    aq1_1   int64  
 2    aq1_2   object 
 3    aq2_1   int64  
 4    aq2_2   object 
 5    aq3_1   int64  
 6    aq3_2   object 
 7    aq4_1   int64  
 8    aq4_2   object 
 9    aq5_1   int64  
 10   aq5_2   object 
 11   aq6_1   int64  
 12   aq6_2   object 
 13   aq7_1   int64  
 14   aq7_2   object 
 15   aq8_1   int64  
 16   aq8_2   object 
 17   aq9_1   int64  
 18   aq9_2   object 
 19   aq10_1  int64  
 20   aq10_2  object 
 21   aq11_1  int64  
 22   aq11_2  object 
 23   aq12_1  int64  
 24   aq12_2  object 
 25   aq13_1  int64  
 26   aq13_2  object 
 27   aq14_1  int64  
 28   aq14_2  object 
 29   aq15_1  int64  
 30   aq15_2  object 
 31   aq16_1  int64  
 32   aq16_2  object 
 33   aq17_1  int64  
 34   aq17_2  object 
 35   aq18_1  int64  
 36   aq18_2  object 
 37   aq19_1  int64  
 38   aq19_2

In [271]:
test_2017.isnull().sum()[:50]

idx       0
aq1_1     0
aq1_2     0
aq2_1     0
aq2_2     0
aq3_1     0
aq3_2     0
aq4_1     0
aq4_2     0
aq5_1     0
aq5_2     0
aq6_1     0
aq6_2     0
aq7_1     0
aq7_2     0
aq8_1     0
aq8_2     0
aq9_1     0
aq9_2     0
aq10_1    0
aq10_2    0
aq11_1    0
aq11_2    0
aq12_1    0
aq12_2    0
aq13_1    0
aq13_2    0
aq14_1    0
aq14_2    0
aq15_1    0
aq15_2    0
aq16_1    0
aq16_2    0
aq17_1    0
aq17_2    0
aq18_1    0
aq18_2    0
aq19_1    0
aq19_2    0
aq20_1    0
aq20_2    0
aq21_1    0
aq21_2    0
aq22_1    0
aq22_2    0
aq23_1    0
aq23_2    0
aq24_1    0
aq24_2    0
aq25_1    0
dtype: int64

In [272]:
test_2017.isnull().sum()[50:100]

aq25_2    0
aq26_1    0
aq26_2    0
aq27_1    0
aq27_2    0
aq28_1    0
aq28_2    0
aq29_1    0
aq29_2    0
aq30_1    0
aq30_2    0
aq31_1    0
aq31_2    0
aq32_1    0
aq32_2    0
aq33_1    0
aq33_2    0
aq34_1    0
aq34_2    0
aq35_1    0
aq35_2    0
aq36_1    0
aq36_2    0
aq37_1    0
aq37_2    0
aq38_1    0
aq38_2    0
aq39_1    0
aq39_2    0
aq40_1    0
aq40_2    0
aq41_1    0
aq41_2    0
bq1       0
bq2       0
bq3       0
bq4       0
bq4_1a    0
bq4_1b    0
bq4_1c    0
bq5       0
bq5_1     0
bq5_2     0
bq6       0
bq7       0
bq8_1     0
bq8_2     0
bq8_3     0
bq9       0
bq10      0
dtype: int64

In [273]:
test_2017.isnull().sum()[100:]

bq11      0
bq12_1    0
bq12_2    0
bq12_3    0
bq12_4    0
bq12_5    0
bq13      0
bq14      0
bq15_1    0
bq15_2    0
bq15_3    0
bq16      0
bq17      0
bq18_1    0
bq18_2    0
bq18_3    0
bq18_4    0
bq18_5    0
bq18_6    0
bq18_7    0
bq19      0
bq19_1    0
bq20      0
bq21      0
bq22      0
bq23      0
bq24_1    0
bq24_2    0
bq24_3    0
bq24_4    0
bq24_5    0
bq24_6    0
bq24_7    0
bq24_8    0
bq25      0
bq26      0
bq27      0
bq28      0
bq29      0
bq30      0
bq31      0
bq32      0
bq33      0
bq34      0
bq35      0
bq36      0
bq37      0
bq38      0
bq38_1    0
bq39_1    0
bq39_2    0
bq40      0
bq41_1    0
bq41_2    0
bq41_3    0
dtype: int64

In [274]:
for i in test_2017.columns : 
    print(i, len(test_2017[i].value_counts()))

idx 9486
aq1_1 5
aq1_2 8
aq2_1 5
aq2_2 8
aq3_1 5
aq3_2 8
aq4_1 5
aq4_2 8
aq5_1 5
aq5_2 8
aq6_1 5
aq6_2 8
aq7_1 5
aq7_2 8
aq8_1 5
aq8_2 8
aq9_1 5
aq9_2 8
aq10_1 5
aq10_2 8
aq11_1 5
aq11_2 8
aq12_1 5
aq12_2 8
aq13_1 5
aq13_2 8
aq14_1 5
aq14_2 8
aq15_1 5
aq15_2 8
aq16_1 5
aq16_2 8
aq17_1 5
aq17_2 8
aq18_1 5
aq18_2 8
aq19_1 5
aq19_2 8
aq20_1 5
aq20_2 8
aq21_1 5
aq21_2 8
aq22_1 5
aq22_2 8
aq23_1 5
aq23_2 8
aq24_1 5
aq24_2 8
aq25_1 5
aq25_2 8
aq26_1 5
aq26_2 8
aq27_1 5
aq27_2 8
aq28_1 5
aq28_2 8
aq29_1 5
aq29_2 8
aq30_1 5
aq30_2 8
aq31_1 5
aq31_2 8
aq32_1 6
aq32_2 8
aq33_1 5
aq33_2 8
aq34_1 5
aq34_2 8
aq35_1 5
aq35_2 8
aq36_1 5
aq36_2 8
aq37_1 5
aq37_2 8
aq38_1 5
aq38_2 8
aq39_1 5
aq39_2 8
aq40_1 5
aq40_2 8
aq41_1 5
aq41_2 8
bq1 21
bq2 6
bq3 6
bq4 2
bq4_1a 1584
bq4_1b 622
bq4_1c 206
bq5 2
bq5_1 7
bq5_2 4011
bq6 7
bq7 7
bq8_1 5
bq8_2 5
bq8_3 5
bq9 5
bq10 5
bq11 5
bq12_1 5
bq12_2 6
bq12_3 6
bq12_4 6
bq12_5 5
bq13 5
bq14 5
bq15_1 5
bq15_2 5
bq15_3 5
bq16 5
bq17 5
bq18_1 5
bq18_2 5
bq18_3 5
bq18

In [275]:
test_2017.aq32_1.value_counts()

1     2639
3     2470
2     1933
4     1915
5      528
33       1
Name: aq32_1, dtype: int64

1. 결측치  
없음

2. 이상치
aq32_1 6 -> 3/3 으로 수정 (확정)
or 1 /빈칸으로 수정 


3. 주관식

bq4_1a 자격증1584
bq4_1b 자격증 622
bq4_1c 자격증 206

버림
bq5_2 훈련사항 4011
bq19_1 일자리변화 8769
bq30 유사직업명 1182
bq31 사용하는도구 6830
bq32 직전직업 1621
bq33 이전가능직업 1429
bq34 신직업발굴 346

bq38_1 학과 1418

4. 숫자여야하는 col

bq23 직업세계변화 51
bq37 나이 63
bq41_1 근로소득 219
bq41_2 근로소득 104
bq41_3 근로소득 153

5. 체크해봐야할 값
bq39_1 고용형태 2
bq39_2 고용형태 7
bq40 계약형태 3

## 이상치 처리하기

In [276]:
test_2017[test_2017.aq32_1==33].idx

7749    7749
Name: idx, dtype: int64

In [277]:
test_2017.iloc[7749, 63] = 3
test_2017.iloc[7749, 64] = 3

In [278]:
test_2017.iloc[7749, 64]

3

In [279]:
for col in test_2017:
        test_2017[col].replace(' ', 0, inplace = True)

In [280]:
test_2017.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9486 entries, 0 to 9485
Data columns (total 155 columns):
 #    Column  Dtype  
---   ------  -----  
 0    idx     int64  
 1    aq1_1   int64  
 2    aq1_2   object 
 3    aq2_1   int64  
 4    aq2_2   object 
 5    aq3_1   int64  
 6    aq3_2   object 
 7    aq4_1   int64  
 8    aq4_2   object 
 9    aq5_1   int64  
 10   aq5_2   object 
 11   aq6_1   int64  
 12   aq6_2   object 
 13   aq7_1   int64  
 14   aq7_2   object 
 15   aq8_1   int64  
 16   aq8_2   object 
 17   aq9_1   int64  
 18   aq9_2   object 
 19   aq10_1  int64  
 20   aq10_2  object 
 21   aq11_1  int64  
 22   aq11_2  object 
 23   aq12_1  int64  
 24   aq12_2  object 
 25   aq13_1  int64  
 26   aq13_2  object 
 27   aq14_1  int64  
 28   aq14_2  object 
 29   aq15_1  int64  
 30   aq15_2  object 
 31   aq16_1  int64  
 32   aq16_2  object 
 33   aq17_1  int64  
 34   aq17_2  object 
 35   aq18_1  int64  
 36   aq18_2  object 
 37   aq19_1  int64  
 38   aq19_2

In [281]:
# 주관식문항 

str_li =['bq4_1a','bq4_1b','bq4_1c','bq5_2', 'bq19_1','bq30','bq31','bq32','bq33','bq34','bq38_1']

In [282]:
for i in test_2017.columns : 
    if i not in str_li : 
        test_2017[i] = test_2017[i].astype(int)
    else : 
        test_2017[i] = test_2017[i].astype(str)

In [283]:
test_2017.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9486 entries, 0 to 9485
Data columns (total 155 columns):
 #    Column  Dtype 
---   ------  ----- 
 0    idx     int32 
 1    aq1_1   int32 
 2    aq1_2   int32 
 3    aq2_1   int32 
 4    aq2_2   int32 
 5    aq3_1   int32 
 6    aq3_2   int32 
 7    aq4_1   int32 
 8    aq4_2   int32 
 9    aq5_1   int32 
 10   aq5_2   int32 
 11   aq6_1   int32 
 12   aq6_2   int32 
 13   aq7_1   int32 
 14   aq7_2   int32 
 15   aq8_1   int32 
 16   aq8_2   int32 
 17   aq9_1   int32 
 18   aq9_2   int32 
 19   aq10_1  int32 
 20   aq10_2  int32 
 21   aq11_1  int32 
 22   aq11_2  int32 
 23   aq12_1  int32 
 24   aq12_2  int32 
 25   aq13_1  int32 
 26   aq13_2  int32 
 27   aq14_1  int32 
 28   aq14_2  int32 
 29   aq15_1  int32 
 30   aq15_2  int32 
 31   aq16_1  int32 
 32   aq16_2  int32 
 33   aq17_1  int32 
 34   aq17_2  int32 
 35   aq18_1  int32 
 36   aq18_2  int32 
 37   aq19_1  int32 
 38   aq19_2  int32 
 39   aq20_1  int32 
 40   aq20

In [284]:

for i in range(41) :
    str5 = 'aq' + str(i+1) +'_1'
    str6 = 'aq' + str(i+1) +'_2'
    
    print(test_2017[(test_2017[str5]==1) & (test_2017[str6]!=0)][str6])

Series([], Name: aq1_2, dtype: int32)
Series([], Name: aq2_2, dtype: int32)
Series([], Name: aq3_2, dtype: int32)
Series([], Name: aq4_2, dtype: int32)
Series([], Name: aq5_2, dtype: int32)
Series([], Name: aq6_2, dtype: int32)
Series([], Name: aq7_2, dtype: int32)
Series([], Name: aq8_2, dtype: int32)
Series([], Name: aq9_2, dtype: int32)
Series([], Name: aq10_2, dtype: int32)
Series([], Name: aq11_2, dtype: int32)
Series([], Name: aq12_2, dtype: int32)
Series([], Name: aq13_2, dtype: int32)
Series([], Name: aq14_2, dtype: int32)
Series([], Name: aq15_2, dtype: int32)
Series([], Name: aq16_2, dtype: int32)
Series([], Name: aq17_2, dtype: int32)
Series([], Name: aq18_2, dtype: int32)
Series([], Name: aq19_2, dtype: int32)
Series([], Name: aq20_2, dtype: int32)
Series([], Name: aq21_2, dtype: int32)
Series([], Name: aq22_2, dtype: int32)
Series([], Name: aq23_2, dtype: int32)
Series([], Name: aq24_2, dtype: int32)
Series([], Name: aq25_2, dtype: int32)
Series([], Name: aq26_2, dtype: in

In [285]:
for i in range(41) :
    str5 = 'aq' + str(i+1) +'_1'
    str6 = 'aq' + str(i+1) +'_2'
    
    print(test_2017[(test_2017[str5].isin([2,3,4,5])) & (test_2017[str6]==0)][str6])

Series([], Name: aq1_2, dtype: int32)
Series([], Name: aq2_2, dtype: int32)
Series([], Name: aq3_2, dtype: int32)
Series([], Name: aq4_2, dtype: int32)
Series([], Name: aq5_2, dtype: int32)
Series([], Name: aq6_2, dtype: int32)
Series([], Name: aq7_2, dtype: int32)
Series([], Name: aq8_2, dtype: int32)
Series([], Name: aq9_2, dtype: int32)
Series([], Name: aq10_2, dtype: int32)
4319    0
5290    0
5960    0
6775    0
6877    0
7350    0
Name: aq11_2, dtype: int32
Series([], Name: aq12_2, dtype: int32)
Series([], Name: aq13_2, dtype: int32)
723     0
1234    0
1452    0
2026    0
3083    0
5078    0
6270    0
7139    0
7320    0
Name: aq14_2, dtype: int32
Series([], Name: aq15_2, dtype: int32)
Series([], Name: aq16_2, dtype: int32)
Series([], Name: aq17_2, dtype: int32)
Series([], Name: aq18_2, dtype: int32)
Series([], Name: aq19_2, dtype: int32)
Series([], Name: aq20_2, dtype: int32)
Series([], Name: aq21_2, dtype: int32)
Series([], Name: aq22_2, dtype: int32)
Series([], Name: aq23_2, 

In [286]:
# aq11_2

to4_li = test_2017[(test_2017.aq11_1.isin([2,3,4,5])) & (test_2017.aq11_2==0)]['aq11_2'].index

In [287]:
to4_li

Int64Index([4319, 5290, 5960, 6775, 6877, 7350], dtype='int64')

In [288]:
train_2017.aq11_2.value_counts()

0    2741
4    1605
5    1414
3    1366
2    1060
6     830
7     314
1     156
Name: aq11_2, dtype: int64

In [289]:
# train에서 0을 제외한 최빈값인 4를 사용

test_2017.iloc[to4_li, 22] =4

In [290]:
to4_li = test_2017[(test_2017.aq11_1.isin([2,3,4,5])) & (test_2017.aq11_2==0)]['aq11_2'].index
to4_li

Int64Index([], dtype='int64')

In [291]:
#aq14_1

to4_li2 =test_2017[(test_2017.aq14_1.isin([2,3,4,5])) & (test_2017.aq14_2==0)].index
to4_li2

Int64Index([723, 1234, 1452, 2026, 3083, 5078, 6270, 7139, 7320], dtype='int64')

In [292]:
train_2017.aq14_2.value_counts()

4    2196
5    1731
3    1604
0    1547
2    1138
6     918
7     204
1     148
Name: aq14_2, dtype: int64

In [293]:
# train 에서 최빈값인 4를 사용
test_2017.iloc[to4_li2, 28]=4

In [294]:
to4_li2 = test_2017[(test_2017.aq14_1.isin([2,3,4,5])) & (test_2017.aq14_2==0)].index
to4_li2

Int64Index([], dtype='int64')

In [295]:
to4_li3  = test_2017[(test_2017.aq27_1.isin([2,3,4,5])) & (test_2017.aq27_2==0)].index
to4_li3

Int64Index([2302, 2479, 2667, 3334, 6279, 8052], dtype='int64')

In [296]:
train_2017.aq27_2.value_counts()

4    2342
3    1830
0    1692
5    1544
2    1188
6     620
1     151
7     119
Name: aq27_2, dtype: int64

In [297]:
# train 에서 최빈값인 4를 사용
test_2017.iloc[to4_li3, 54]=4

In [298]:
to4_li3  =test_2017[(test_2017.aq27_1.isin([2,3,4,5])) & (test_2017.aq27_2==0)].index
to4_li3

Int64Index([], dtype='int64')

In [299]:
to4_li4=test_2017[(test_2017.aq32_1.isin([2,3,4,5])) & (test_2017.aq32_2==0)].index
to4_li4

Int64Index([1362, 1651, 2854, 3194, 3216, 3280, 8775, 8811, 8853], dtype='int64')

In [300]:
train_2017.aq32_2.value_counts()

0    2643
4    1754
5    1517
3    1414
2    1064
6     654
7     228
1     212
Name: aq32_2, dtype: int64

In [301]:
# train에서 0을 제외한 최빈값인 4를 사용
test_2017.iloc[to4_li4, 64]=4

In [302]:
to4_li4=test_2017[(test_2017.aq32_1.isin([2,3,4,5])) & (test_2017.aq32_2==0)].index
to4_li4

Int64Index([], dtype='int64')

In [303]:
to4_li5=test_2017[(test_2017.aq41_1.isin([2,3,4,5])) & (test_2017.aq41_2==0)].index
to4_li5

Int64Index([1284, 3460, 5419, 6893, 8305], dtype='int64')

In [304]:
train_2017.aq41_2.value_counts()

0    3582
3    1593
4    1443
2    1351
5     916
6     279
1     256
7      66
Name: aq41_2, dtype: int64

In [305]:
# train에서 0을 제외한 최빈값인 4를 사용
test_2017.iloc[to4_li5, 82]=4

In [306]:
to4_li5=test_2017[(test_2017.aq41_1.isin([2,3,4,5])) & (test_2017.aq41_2==0)].index
to4_li5

Int64Index([], dtype='int64')

In [307]:
## 근로소득 처리 
#test 결측치를 train의 업종의 중위값으로 대체 


In [308]:
null_li = test_2017[(test_2017.bq39_1==1) & (test_2017.bq41_1==0)].index
null_li

Int64Index([ 117,  582,  848,  930,  980, 1125, 1175, 1365, 1449, 1653, 1664,
            2163, 2874, 4216, 4589, 4597, 5030, 5063, 5196, 5392, 5530, 5701,
            5783, 5973, 6007, 6463, 6514, 6996, 7319, 7455, 7713, 8198, 8264,
            8430, 8740, 8762, 8954, 9201],
           dtype='int64')

In [309]:
for i in null_li:
     test_2017.iloc[i,152]=round(train_2017[(train_2017.bq1 ==test_2017.iloc[i,83])&(train_2017.bq41_1!=0)].bq41_1.median())

In [310]:
null_li = test_2017[(test_2017.bq39_1==1) & (test_2017.bq41_1==0)].index
null_li

Int64Index([], dtype='int64')

In [311]:
null_li2 =test_2017[(test_2017.bq39_1==1) & (test_2017.bq41_3==0)].index
null_li2

Int64Index([980, 1175, 1653, 2874, 3194, 5241, 6402, 7405, 8954], dtype='int64')

In [312]:
for i in null_li2:
    test_2017.iloc[i,154]=round(train_2017[(train_2017.bq1 ==test_2017.iloc[i,83])&(train_2017.bq41_3!=0)].bq41_3.median())
    

In [313]:
null_li2 =test_2017[(test_2017.bq39_1==1) & (test_2017.bq41_3==0)].index
null_li2

Int64Index([], dtype='int64')

In [314]:
null_li3 = test_2017[(test_2017.bq39_1==2) & (test_2017.bq41_2==0)].index
null_li3 

Int64Index([ 121,  218,  288, 1290, 1301, 1324, 1529, 2389, 2633, 2974, 3171,
            3573, 3901, 4068, 4351, 4667, 4751, 5040, 5339, 5359, 5394, 5911,
            5993, 5994, 6156, 6388, 6723, 7261, 7694, 7715, 7809, 8412, 8641,
            8676, 8781, 8987, 9078, 9258],
           dtype='int64')

In [315]:
for i in null_li3:
    test_2017.iloc[i,153]=round(train_2017[(train_2017.bq1 ==test_2017.iloc[i,83])&(train_2017.bq41_2!=0)].bq41_2.median())
    

In [316]:
null_li3 = test_2017[(test_2017.bq39_1==2) & (test_2017.bq41_2==0)].index
null_li3 

Int64Index([], dtype='int64')

In [317]:
## 자격증 없음 없다 처리 

for i in test_2017.bq4_1b:
    if '없' in i : 
        print(i)

        
test_2017['bq4_1b'].replace('없다', '0', inplace = True)


없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다


In [318]:
# 처리되었는지 확인
for i in test_2017.bq4_1b:
    if '없' in i :
        print(i)

In [319]:
for i in test_2017.bq4_1c:
    if '없' in i : 
        print(i)

        
test_2017['bq4_1c'].replace('없다', '0', inplace = True)

없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다
없다


In [320]:
for i in test_2017.bq4_1c:
    if '없' in i : 
        print(i)


In [321]:
test_2017[(test_2017.bq38==1) & (test_2017.bq38_1!='0')]['bq38_1'].value_counts()

Series([], Name: bq38_1, dtype: int64)

In [322]:
test_2017[(test_2017.bq38.isin([2,3,4,5,6])) & (test_2017.bq38_1=='0')]['bq38_1'].value_counts()

Series([], Name: bq38_1, dtype: int64)

In [323]:
## 학과 없음, 없다 처리 

for i in test_2017.bq38_1:
    if '없' in i : 
        print(i)
        
test_2017['bq38_1'].replace('없다', '0', inplace = True)

없다
없다


In [324]:
for i in test_2017.bq38_1:
    if '없' in i : 
        print(i)

In [325]:
test_2017.to_csv("know_2017_test_enutf.csv", encoding='utf8', index=False)
test_2017.to_csv("know_2017_test_encp.csv", encoding='cp949', index=False)

In [326]:
df_cp = pd.read_csv("know_2017_test_encp.csv", encoding='cp949')
df_cp.info(verbose=1)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9486 entries, 0 to 9485
Data columns (total 155 columns):
 #    Column  Dtype 
---   ------  ----- 
 0    idx     int64 
 1    aq1_1   int64 
 2    aq1_2   int64 
 3    aq2_1   int64 
 4    aq2_2   int64 
 5    aq3_1   int64 
 6    aq3_2   int64 
 7    aq4_1   int64 
 8    aq4_2   int64 
 9    aq5_1   int64 
 10   aq5_2   int64 
 11   aq6_1   int64 
 12   aq6_2   int64 
 13   aq7_1   int64 
 14   aq7_2   int64 
 15   aq8_1   int64 
 16   aq8_2   int64 
 17   aq9_1   int64 
 18   aq9_2   int64 
 19   aq10_1  int64 
 20   aq10_2  int64 
 21   aq11_1  int64 
 22   aq11_2  int64 
 23   aq12_1  int64 
 24   aq12_2  int64 
 25   aq13_1  int64 
 26   aq13_2  int64 
 27   aq14_1  int64 
 28   aq14_2  int64 
 29   aq15_1  int64 
 30   aq15_2  int64 
 31   aq16_1  int64 
 32   aq16_2  int64 
 33   aq17_1  int64 
 34   aq17_2  int64 
 35   aq18_1  int64 
 36   aq18_2  int64 
 37   aq19_1  int64 
 38   aq19_2  int64 
 39   aq20_1  int64 
 40   aq20

In [327]:
df_utf = pd.read_csv("know_2017_test_enutf.csv")
df_utf.info(verbose=1)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9486 entries, 0 to 9485
Data columns (total 155 columns):
 #    Column  Dtype 
---   ------  ----- 
 0    idx     int64 
 1    aq1_1   int64 
 2    aq1_2   int64 
 3    aq2_1   int64 
 4    aq2_2   int64 
 5    aq3_1   int64 
 6    aq3_2   int64 
 7    aq4_1   int64 
 8    aq4_2   int64 
 9    aq5_1   int64 
 10   aq5_2   int64 
 11   aq6_1   int64 
 12   aq6_2   int64 
 13   aq7_1   int64 
 14   aq7_2   int64 
 15   aq8_1   int64 
 16   aq8_2   int64 
 17   aq9_1   int64 
 18   aq9_2   int64 
 19   aq10_1  int64 
 20   aq10_2  int64 
 21   aq11_1  int64 
 22   aq11_2  int64 
 23   aq12_1  int64 
 24   aq12_2  int64 
 25   aq13_1  int64 
 26   aq13_2  int64 
 27   aq14_1  int64 
 28   aq14_2  int64 
 29   aq15_1  int64 
 30   aq15_2  int64 
 31   aq16_1  int64 
 32   aq16_2  int64 
 33   aq17_1  int64 
 34   aq17_2  int64 
 35   aq18_1  int64 
 36   aq18_2  int64 
 37   aq19_1  int64 
 38   aq19_2  int64 
 39   aq20_1  int64 
 40   aq20